### DISCUSSION TITLE
Required text type when generating models via CLI

### DISCUSSION THREAD

**Discussion by yinho999** [loco-rs/loco#262] [Q&A]
I am wondering if is there any reason why the required text type is not a thing in the CLI when creating the model using the CLI tool?

**Comment by kaplanelad** [comment]
@yinho999 can you elaborate? not sure I'm understand your question

**Reply by yinho999** [reply]
Sorry for my bad description. There is "string!" for string not null and "int!" for int no null but "text!" is not available in the CLI. I am thinking is there a special reason for that.

**Comment by kaplanelad** [comment]
the `string` is defined as the column type, in this case, varchar.
for example if I run the command:
```sh
cargo loco generate model article title:string title2:string!
```

The column `title` will created and this field is a must to set when you create a new record
The column `title2` with the `!` will not be a mandatory field and this field can be null
